pkg/web/apps/apps_psql_repository.go

```diff
@@ -128,5 +130,154 @@ func (a *appsPostgreSQLRepository) GetAppByID(ID string) (*models.App, error) {
 	}
 
 	return &app, nil
+}
+
+// GetDeviceByDeviceID gets a device by its Device ID
+func (a *appsPostgreSQLRepository) GetDeviceByDeviceID(deviceID string) (*models.Device, error) {
+	query := `
+	SELECT id,version_id,app_id,device_id,device_type,device_version
+	FROM device
+	WHERE device_id = $1;`
+
+	row := a.db.QueryRow(query, deviceID)
+
+	var device models.Device
+	if err := row.Scan(&device.ID, &device.VersionID, &device.AppID, &device.DeviceID, &device.DeviceType, &device.DeviceVersion); err != nil {
+		log.Error(err)
+		switch err {
+		case sql.ErrNoRows:
+			return nil, models.ErrNotFound
+		default:
+			return nil, models.ErrDatabaseError
+		}
+	}
+
+	return &device, nil
+}
+
+// GetVersionByAppIDAndVersion gets a version by its app ID and version number
+func (a *appsPostgreSQLRepository) GetVersionByAppIDAndVersion(appID string, versionNumber string) (*models.Version, error) {
+	version := models.Version{}
+	var disabledMessage sql.NullString
+	var lastLaunchedAt sql.NullString
```

Comment:
If a version exists, this `lastLaunchedAt` will be set. There should never be a scenario where it is null in the database in my opinion.